sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -245,14 +246,26 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
       query: LogicalPlan): LogicalPlan = withOrigin(ctx) {
     import ctx._
 
+    // Replaces ordinal in 'order by' or 'sort by' with UnresolvedOrdinal expression
+    def replacesOrdinal(orders: Seq[SortOrder]): Seq[SortOrder] = {
```

Comment:
Because we need to apply the same on Dataset API. We may need to extract this out so we can reuse it to Dataset API.